core/src/HIP/Kokkos_HIP_ParallelFor_Range.hpp

```diff
@@ -58,14 +58,13 @@ class ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>, Kokkos::HIP> {
   ParallelFor& operator=(ParallelFor const&) = delete;
 
   inline __device__ void operator()() const {
-    const Member work_stride = blockDim.y * gridDim.x;
-    const Member work_end    = m_policy.end();
+    const auto work_stride = Member(blockDim.y) * gridDim.x;
+    const Member work_end  = m_policy.end();
 
     for (Member iwork =
              m_policy.begin() + threadIdx.y + blockDim.y * blockIdx.x;
          iwork < work_end;
-         iwork = iwork < work_end - work_stride ? iwork + work_stride
-                                                : work_end) {
+         iwork = std::min<Member>(iwork + work_stride, work_end)) {
```

Comment:
I think you might overflow with this change. I think it was purposefully that it is a ternary, to ensure that you never evaluate `iwork + work_stride` when it might overflow. Am I wrong?

Reply:
@maartenarnst ?